dora/core/server/worker/src/main/java/alluxio/worker/DataServerFactory.java

```diff
@@ -59,9 +60,9 @@ protected DataServerFactory(UfsManager ufsManager,
    */
   public DataServer createRemoteGrpcDataServer(DataWorker dataWorker) {
     BlockWorkerGrpc.BlockWorkerImplBase blockWorkerService;
-    if (dataWorker instanceof DoraWorker) {
+    if (dataWorker instanceof PagedDoraWorker) {
       blockWorkerService =
-          new DoraWorkerClientServiceHandler((DoraWorker) dataWorker);
+          new DoraWorkerClientServiceHandler((PagedDoraWorker) dataWorker);
```

Comment:
`DoraWorker` interface is missing a lot of method definitions so we need to downcast in `DoraWorkerClientServiceHandler` after all. So I'm just casting to the real class to avoid the frequent casting.